includes/woopay/class-woopay-session.php

```diff
@@ -46,6 +52,44 @@ public static function init() {
 		add_filter( 'determine_current_user', [ __CLASS__, 'determine_current_user_for_woopay' ], 20 );
 		add_filter( 'rest_request_before_callbacks', [ __CLASS__, 'add_woopay_store_api_session_handler' ], 10, 3 );
 		add_action( 'woocommerce_store_api_checkout_update_order_meta', [ __CLASS__, 'remove_order_customer_id_on_requests_with_verified_email' ] );
+
+		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'init_woopay_session_data_button' ], 20 );
+		add_action( 'wp_head', [ __CLASS__, 'init_woopay_session_data_checkout' ] );
+	}
+
+	/**
+	 * Initializes the session data for WooPay to be combined with the /verify request.
+	 * This is only used for the express button outside of the checkout page.
+	 *
+	 * @return void
+	 */
+	public static function init_woopay_session_data_button() {
+
+		// the checkout is excluded and handled in init_woopay_session_data_checkout.
+		if ( ! WC_Payments_Features::is_woopay_enabled() || is_checkout() ) {
+			return;
+		}
+
+		$session_data = self::get_init_session_request();
+
+		wp_add_inline_script( 'WCPAY_WOOPAY_EXPRESS_BUTTON', 'sessionDataWooPay=' . wp_json_encode( $session_data ), 'before' );
+	}
+
+	/**
+	 * Initializes the session data for WooPay to be combined with the /verify request.
+	 * This is only used on the checkout page.
+	 *
+	 * @return void
+	 */
+	public static function init_woopay_session_data_checkout() {
+
+		if ( ! WC_Payments_Features::is_woopay_enabled() || ! is_checkout() ) {
+			return;
+		}
+
+		$session_data = self::get_init_session_request();
+
+		echo '<script type="text/javascript">sessionDataWooPay=' . wp_json_encode( $session_data ) . '</script>';
```

Comment:
Why do we need two different methods to add the data to the frontend?

Reply:
The logic was changed on https://github.com/Automattic/woocommerce-payments/pull/6831/commits/1afcf02fdb1f67f21b94fbd1289b2d49ac6e1989 and the session data is not printed in the page anymore.